utils/python/CIME/hist_utils.py

```diff
@@ -0,0 +1,172 @@
+"""
+Functions for actions pertaining to history files.
+"""
+
+from CIME.XML.standard_module_setup import *
+from CIME.XML.component import Component
+
+import logging, glob, os, shutil, re
+
+def _iter_model_file_substrs():
+    drv_comp = Component()
+    models = drv_comp.get_valid_model_components()
+
+    for model in models:
+        model = "cpl" if model == "DRV" else model.lower()
+        yield model
+
+def _get_all_hist_files(testcase, model, suffix="", from_dir=os.getcwd()):
+    suffix = (".%s" % suffix) if suffix else ""
+    test_hists = glob.glob("%s/%s.%s*.h?.nc%s" % (from_dir, testcase, model, suffix))
```

Comment:
> Right now it's comparing all of the history files - I would suggest that it would be better to solve this by reducing the history files written in the test instead of hard coding something here.

I agree, in principle. However, off-hand, I'm having trouble seeing an easy way to do that. Two options I can think of are:

1. Giving components the option to listen to the coupler's HIST_OPTION in determining their history frequency. I have done this for CISM, but I don't think any other CESM components have this ability. I discussed this with @mvertens (and maybe you?) a while ago, and we felt this would be non-trivial.

2. Using testmods. However, most tests in the CLM test suite already have testmods to control their options, so adding separate testmods to control history frequency would lead to an explosion of testmods - e.g., in addition to the existing crop and crop_monthly testmods that we have, we'd need crop_annual, crop_5years, crop_20years, etc. The best way I could see to handle this would be to allow a test to specify two different testmods that are both applied... if that's feasible, that could be a good solution.

Thoughts?